include/oneapi/dpl/pstl/utils_ranges.h

```diff
@@ -779,81 +779,59 @@ __get_subscription_view(_Range&& __rng)
     return std::forward<_Range>(__rng);
 }
 
-template <typename _Source, typename _Base = std::decay_t<_Source>>
-struct __subscription_impl_view_simple : _Base
+#if _ONEDPL_CPP20_RANGES_PRESENT
+
+template <std::ranges::view _View>
+    requires std::ranges::view<_View> && std::ranges::random_access_range<_View>
+struct __subscription_impl_view_simple : std::ranges::view_interface<__subscription_impl_view_simple<_View>>
 {
     static_assert(
-        !__has_subscription_op<_Base>::value,
-        "The usage of __subscription_impl_view_simple prohibited if std::decay_t<_Source>::operator[] implemented");
+        !__has_subscription_op<_View>::value,
+        "The usage of __subscription_impl_view_simple prohibited if _View::operator[] implemented");
 
-    using value_type = oneapi::dpl::__internal::__value_t<_Base>;
-    using index_type = oneapi::dpl::__internal::__difference_t<_Base>;
+    _View __base;
 
-    // Define default constructors
-    __subscription_impl_view_simple(const __subscription_impl_view_simple&) = default;
-    __subscription_impl_view_simple(__subscription_impl_view_simple&&) = default;
-
-    // Define custom constructor to forward arguments to the base class
-    template <typename... _Args>
-    __subscription_impl_view_simple(_Args&&... __args) : _Base(std::forward<_Args>(__args)...)
+    constexpr explicit __subscription_impl_view_simple(_View __view) : __base(std::move(__view))
     {
     }
 
-    // Define default operator=
-    __subscription_impl_view_simple&
-    operator=(const __subscription_impl_view_simple&) = default;
-    __subscription_impl_view_simple&
-    operator=(__subscription_impl_view_simple&&) = default;
-
-    // We need to define these begin(), end() and size() functions to redirect the ADL mechanism
-    // from our (enclosing) class to the corresponding call for our base class
     constexpr auto
     begin() const
     {
-        return __begin(get_base_ref());
+        return oneapi::dpl::__ranges::__begin(base());
     }
 
     constexpr auto
     end() const
     {
-        return __end(get_base_ref());
+        return oneapi::dpl::__ranges::__end(base());
     }
 
-    auto
+    constexpr auto
     size() const
     {
-        return __size(get_base_ref());
-    }
-
-    decltype(auto)
-    operator[](index_type __i) const
-    {
-        assert(__i < size());
-        return *std::next(__begin(get_base_ref()), __i);
+        return oneapi::dpl::__ranges::__size(base());
```

Comment:
I don't think these gain anything from being `constexpr`.  The functions they call are not.